mail_forward/static/src/js/mail_forward.js

```diff
@@ -2,76 +2,122 @@
+                    _t("Date: ") + message.date
+                ];
+                if (message.subject) {
+                    header.push(_t("Subject: ") + message.subject);
+                }
+                if (message.email_to) {
+                    header.push(_t("To: ") + message.email_to);
+                }
+                if (message.email_cc) {
+                    header.push(_t("CC: ") + message.email_cc);
+                }
+                header = header.map(_.str.escapeHTML).join("<br/>");
 
-            var context = {
-                default_attachment_ids: attachment_ids,
-                default_body:
+                var context = {
+                    default_attachment_ids: attachment_ids,
+                    default_body:
                     "<p><i>" + header + "</i></p><br/>" +
-                    this.body,
-                default_model: this.model,
-                default_res_id: this.res_id,
-                default_subject: subject.join(": "),
-            };
+                    message.body,
+                    default_model: message.model,
+                    default_res_id: message.res_id,
+                    default_subject: subject.join(": ")
+                };
 
-            if (this.model && this.res_id) {
-                context.default_destination_object_id =
-                    [this.model, this.res_id].join();
-            }
+                if (message.model && message.res_id) {
+                    context.default_destination_object_id =
+                        [message.model, message.res_id].join();
+                }
 
-            // Get the action data and execute it to open the composer wizard
-            var do_action = this.do_action;
-            this.rpc("/web/action/load", {
-                "action_id": "mail_forward.compose_action",
-            })
-            .done(function(action) {
-                action.context = context;
-                do_action(action);
+                // Get the action data and execute it to open the composer wizard
+                var do_action = self.do_action;
+                self.rpc("/web/action/load", {
+                    "action_id": "mail_forward.compose_action"
+                })
+                    .done(function (action) {
+                        action.context = context;
+                        do_action(action, {
+                            on_close: $.proxy(self.thread_reload, self)
+                        });
+                    });
+            });
+        },
+        init: function (parent, options) {
+            this._super.apply(this, arguments);
+            // Add click reaction in the events of the thread object
+            this.events['click .o_forward'] = function (event) {
+                var message_id = $(event.currentTarget).data('message-id');
+                this.trigger("message_forward", message_id);
+            };
+        },
+        thread_reload: function(){
```

Comment:
I'm not sure I like this new feature.

I did a test from the Odoo inbox forwarding to a different object... If you are forwarding to a different object, why to reload? It should never get displayed...

It moves me above in the message list if I was browsing somewhere below, and clears the messages. Well... it's a confusing behavior.

Reply:
@rafaelbn originally requested this feature. As I said, I'm no javascript expert so maybe I did not implement it the best way. Anyway I would need help to improve javascript.